### ISSUE TITLE
SHEETPATH variable in KiCad V7 not correct across hierarchies

### ISSUE THREAD

**Issue by myamigo** [openscopeproject/InteractiveHtmlBom#358]
I want using this plugin to aggregate capacitance and inductance within each hierarchal block and figured I could use add the ${SHEETPATH} variable, which is now supported in KiCad V7, to the symbol as follows:

![image](https://user-images.githubusercontent.com/17319745/219662277-06671a7f-db9b-48e4-833c-51e5516a543f.png)

The field is properly evaluated in the 3 hierarchical sheets where it is used... 
![image](https://user-images.githubusercontent.com/17319745/219662214-9f4d5628-479e-4930-b3a0-b3f9e47b9ff7.png)

I intend to parse the SHEETPATH output to identify the root sheet value (data_path#) for all components nested under that sheet.

With the field added to the symbols I would then add the new symbol field name to the **ibom.config.ini** file. This goes in the [fields] section as follows...

```
[fields]
show_fields=Value,Footprint,Sheet_Path
group_fields=Value,Footprint,Sheet_Path
 ```

I selected the option to group by the new field as well so this should give me three lines for each common part but it does not... 

![image](https://user-images.githubusercontent.com/17319745/219663821-f4aea9ca-0997-421f-ab73-464c1df84906.png)

So I changed the settings to produce an ungrouped BOM... 

![image](https://user-images.githubusercontent.com/17319745/219662601-839a69b5-14a2-4240-827d-29280ad0df72.png)

... to see what the script found...

![image](https://user-images.githubusercontent.com/17319745/219664278-8ca65f62-33a6-40b5-b5d4-23eb8759ffe6.png)

... but rather than seeing 3 lines for the part with each indicating a different _**data_path#**_ root, I see that they are all the same (all based on data_path0). The 2nd & 3rd lines should indicate data_path1 & data_path2 respectively. 

We know that these fields are rendered correctly in KiCad so this must be an issue with the plugin.

**Comment by qu1ck** [comment]
What are you using as extra data source?
![image](https://user-images.githubusercontent.com/3622532/219678473-e3f0e3e4-6c10-4afd-86f6-48c8ad792169.png)

If it's the pcb file then you need to pull changes from schematic before generating ibom.
![image](https://user-images.githubusercontent.com/3622532/219679130-1369d742-7001-4daf-9420-40040eb40b95.png)

If it's a netlist file then you need to regenerate netlist.

**Comment by myamigo** [comment]
I am using the PCB and I am pulling from the schematic.

**Comment by qu1ck** [comment]
Yeah, I just tested this and it's a bug in kicad. If you open the pcb file in text editor and search for "sheet_name" you will see it has the same value for all instances of components regardless which sheet they came from.

But the workaround is to use the built in sheetname field instead of adding your own. Does that work for you?

**Comment by myamigo** [comment]
I'm trying to use ${SHEETPATH} rather than ${SHEETNAME}. I'll take a workaround  but I'm not clear on what you're suggesting.

**Comment by myamigo** [comment]
I do see real issues in the PCB file.  Initially I had just 1 part with the Sheet_Path = **SHEETPATH** but I just added another. Now I am seeing the evaluated **SHEETPATH** value from the new part, which is on a different path, showing up in both places. I think I can generate an issue  in KiCad for this.

I understand now that your script uses  the already evaluated contents of the field. I though it was evaluating the SHEETPATH variable on the fly.

Thanks for the quick feedback,

**Comment by qu1ck** [comment]
Pcbnew generates sheetname and few other fields automatically
![image](https://user-images.githubusercontent.com/3622532/219685219-1402a574-9c25-4983-a81e-c749eaa03005.png)

I thought you could use that instead of path but because in your case the different part of path is the portion in the middle and not the end, it won't work.

**Comment by myamigo** [comment]
All good! I know what to do to get this raised as an issue.

**Comment by qu1ck** [comment]
https://gitlab.com/kicad/code/kicad/-/issues/13980

**Comment by myamigo** [comment]
Very good. I'll add my test file to that. It includes the SHEETNAME and SHEETPATH variables in the symbol so you can see the disconnect when used in the PCB.

**Comment by myamigo** [comment]
I do have a solution though. The SHEETPATH variable is supported in the Schematic Editor and the KiBOM python script handles the SHEETPATH variable the way I want so I'll use that for my data analysis and your plugin to make it easy to find parts, which it does very well.

**Comment by qu1ck** [comment]
This is fixed in kicad 7.0.5
